web/src/test/java/org/springframework/security/web/util/matcher/MediaTypeRequestMatcherTests.java

```diff
@@ -102,6 +113,77 @@ public void mediaAllMatches() throws Exception {
 		assertThat(matcher.matches(request)).isTrue();
 	}
 
+	@Test
+	public void matchWhenAcceptHeaderAsteriskThenAll() throws Exception {
+		request.addHeader("Accept", "*/*");
+		matcher = new MediaTypeRequestMatcher(MediaType.ALL);
+		assertThat(matcher.matches(request)).isTrue();
+	}
+
+	@Test
+	public void matchWhenAcceptHeaderAsteriskThenAnyone() throws Exception {
+		request.addHeader("Accept", "*/*");
+		matcher = new MediaTypeRequestMatcher(MediaType.TEXT_HTML);
+		assertThat(matcher.matches(request)).isTrue();
+	}
+
+	@Test
+	public void matchWhenAcceptHeaderAsteriskThenAllInCollection() throws Exception {
```

Comment:
Sorry, I just noticed this myself: This test and the following test are the same. It seems like you intended to have the second test be `MediaType.TEXT_HTML`.

Reply:
sorry, it's a typo, fixed